src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/læremidler/beregning/LæremidlerBeregningService.kt

```diff
@@ -34,15 +40,27 @@ class LæremidlerBeregningService(
         behandlingId: BehandlingId,
     ): BeregningsresultatLæremidler {
         val stønadsperioder = hentStønadsperioder(behandlingId)
+        val behandling = behandlingRepository.finnSaksbehandling(behandlingId)
+        val vedtaksperioderForrigeBehandling = hentForrigeVedtaksperioder(behandling)
 
-        validerVedtaksperioder(vedtaksperioder, stønadsperioder)
+        validerVedtaksperioder(
+            vedtaksperioder,
+            vedtaksperioderForrigeBehandling,
+            stønadsperioder,
+            behandling.revurderFra,
+        )
 
         val aktiviteter = finnAktiviteter(behandlingId)
         val beregningsresultatForMåned = beregnLæremidlerPerMåned(vedtaksperioder, stønadsperioder, aktiviteter)
 
         return BeregningsresultatLæremidler(beregningsresultatForMåned)
     }
 
+    private fun hentForrigeVedtaksperioder(behandling: Saksbehandling?): List<Vedtaksperiode>? =
+        behandling?.forrigeBehandlingId?.let {
+            vedtaksRepository.findByBehandlingId(behandling.forrigeBehandlingId).finnVedtaksperioder()
+        }
```

Comment:
Koden er egentlige riktig, men det går også å bruke noen metoder vi har, blant annet `vedtakService.hentVedtak<VedtakLæremidler>`
```suggestion
    private fun hentForrigeVedtaksperioder(behandling: Saksbehandling?): List<Vedtaksperiode>? =
        behandling?.forrigeBehandlingId?.let {
            vedtakService.hentVedtak<VedtakLæremidler>(it)?.data?.vedtaksperioder()
        }
```
Vid en revurdering kommer "forrigeBehandlingId" alltid peke til en behandling som er iverksatt, dvs innvilgelse eller opphør, men ikke avslag eller henlagt, så går det også å bruke
```kotlin
private fun hentForrigeVedtaksperioder(behandling: Saksbehandling?): List<Vedtaksperiode>? =
        behandling?.forrigeBehandlingId?.let {
            vedtakService.hentVedtak<InnvilgelseEllerOpphørLæremidler>(it)?.data?.vedtaksperioder
        }
```

Reply:
Nice! liker godt å bruke servicene. Endrer til det 👍